### ISSUE TITLE
possible PLC2701 false positive

### ISSUE THREAD

**Issue by dotysan** [astral-sh/ruff#22187]
### Summary

Apolgies if I'm misunderstanding. But this:
```py
from mirror_syno.__main__ import main
```
is throwing PLC2701 on the last word 'main':
```
Private name import `__main__` from external module `mirror_syno`
```

There is no dunder in `main`. And I don't believe `__main__` as a module is a violation.

**Comment by ntBre** [comment]
Thanks for the report! Based on the rule test suite, I think flagging private sub-modules is intentional:

https://github.com/astral-sh/ruff/blob/139149f87b4ad3afb1fe2f4b97af280449fdd741/crates/ruff_linter/src/rules/pylint/snapshots/ruff_linter__rules__pylint__tests__PLC2701_import_private_name__submodule____main__.py.snap#L14-L20

but I think it would make sense both to have an exception for dunder names, which we have for the root module [here](https://github.com/astral-sh/ruff/blob/139149f87b4ad3afb1fe2f4b97af280449fdd741/crates/ruff_linter/src/rules/pylint/rules/import_private_name.rs#L99), and also to make the diagnostic range on the actual private name, as you said.
